src/app/utils/field-validation/validators/childrenValidator.ts

```diff
@@ -47,22 +51,34 @@ const validChildAnswerFirstArray: ChildrenValidator = (response) => {
 const validChildAnswerConsistency: ChildrenValidator = (response) => {
   const { answerArray } = response
   const len = answerArray[0].length
+
   return answerArray.every((subArr) => subArr.length === len)
     ? right(response)
-    : left(`ChildrenValidator:\t inconsistent answer array subarrays`)
+    : left(
+        `ChildrenValidator (validChildAnswerConsistency):\t inconsistent answer array subarrays`,
+      )
 }
 
 /**
  * Returns a validation function to check if
- * all the answers are non-empty.
+ * all the answers are non-empty if first answer subarray has length > 0 and a child is selected
  */
 const validChildAnswersNonEmpty: ChildrenValidator = (response) => {
-  const { answerArray } = response
-  return answerArray.every((subArr) =>
-    subArr.every((val) => typeof val === 'string' && !!val.trim()),
-  )
-    ? right(response)
-    : left(`ChildrenValidator:\t inconsistent answer array subarrays`)
+  const { childSubFieldsArray, answerArray } = response
+  const noOfChildrenSubFields = childSubFieldsArray?.length ?? 1
+  const noChildSelectedAnswerArray = Array(noOfChildrenSubFields).fill('')
+  const first = answerArray[0]
+  return Array.isArray(first) &&
+    first.length > 0 &&
+    first[0] !== noChildSelectedAnswerArray[0]
```

Comment:
cmiiw, this is to check if `first[0]` is an empty field? If that's the case, does it make sense if we have it as `first[0] !== ''`?
 
I was initially confused with why we are comparing the first elements of both arrays.

Reply:
This is to check if a child is selected. I added more comments in [d01885f](https://github.com/opengovsg/FormSG/pull/6673/commits/d01885f12aa742686227ba20c4dafa4908791a0b), hope it's slightly clearer now!

first[0] is the answer from the response, and `noChildSelectedAnswer` is the format that the answer should take if no child is selected (a string of empty strings). We want to check if at least 1 child is selected, before we do the validation.

Reply:
Got it! 👍